pom.xml

```diff
@@ -60,6 +70,45 @@
 					<target>1.8</target>
 				</configuration>
 			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
+				<version>3.1.1</version>
+				<configuration>
+					<dependencyReducedPomLocation>${project.build.directory}/dependency-reduced-pom.xml</dependencyReducedPomLocation>
+					<relocations>
+						<relocation>
+							<pattern>io.papermc.lib</pattern>
+							<shadedPattern>com.wimbli.WorldBorder.paperlib</shadedPattern> <!-- Replace this -->
+						</relocation>
+					</relocations>
+				</configuration>
+				<executions>
+					<execution>
+						<phase>package</phase>
+						<goals>
+							<goal>shade</goal>
+						</goals>
+						<configuration>
+							<artifactSet>
+							  <excludes>
+								<exclude>org.spigotmc:spigot-api</exclude>
+								<exclude>commons-lang:commons-lang</exclude>
+								<exclude>com.googlecode.json-simple:json-simple</exclude>
+								<exclude>junit:junit</exclude>
+								<exclude>org.hamcrest:hamcrest-core</exclude>
+								<exclude>com.google.guava:guava</exclude>
+								<exclude>com.google.code.gson:gson</exclude>
+								<exclude>org.yaml:snakeyaml</exclude>
+								<exclude>net.md-5:bungeecord-chat</exclude>
+								<exclude>org.bukkit:bukkit</exclude>
+								<exclude>us.dynmap:dynmap-api</exclude>
+							  </excludes>
```

Comment:
This list could be replaced with includes for the specific libraries that should be shaded in. Besides shortening the pom it would also avoid possible future issues where new libraries are depended on that didn't get added to the exclude list and then shaded in.

Reply:
Well .. the real issue is, the original pom.xml didn't set the scope of dependencies, so `compile` was assumed. The maven shader includes dependencies that have `compile` scope, and omits `provided` scope dependences, in accordance with their meaning (compile => I need to take care of them vs. provided => I can assume these are there). So I removed all excludes, set the scope of all dependencies except PaperLib to provided, and voila! only PaperLib gets shaded.